waspc/src/Wasp/Generator/WebAppGenerator.hs

```diff
@@ -246,16 +247,28 @@ genIndexJs spec =
       (C.asWebAppFile [relfile|src/index.js|])
       ( Just $
           object
-            [ "doesClientSetupFnExist" .= isJust maybeSetupJsFunction,
-              "clientSetupJsFnImportStatement" .= fromMaybe "" maybeSetupJsFnImportStmt,
-              "clientSetupJsFnIdentifier" .= fromMaybe "" maybeSetupJsFnImportIdentifier
+            [ "setupFn" .= mkTmplDataForExtImport maybeSetupJsFunction,
+              "rootComponent" .= mkTmplDataForExtImport maybeRootComponent
             ]
       )
   where
     maybeSetupJsFunction = AS.App.Client.setupFn =<< AS.App.client (snd $ getApp spec)
-    maybeSetupJsFnImportDetails = getJsImportStmtAndIdentifier relPathToWebAppSrcDir <$> maybeSetupJsFunction
-    (maybeSetupJsFnImportStmt, maybeSetupJsFnImportIdentifier) =
-      (fst <$> maybeSetupJsFnImportDetails, snd <$> maybeSetupJsFnImportDetails)
+    maybeRootComponent = AS.App.Client.rootComponent =<< AS.App.client (snd $ getApp spec)
 
-    relPathToWebAppSrcDir :: Path Posix (Rel ()) (Dir C.WebAppSrcDir)
-    relPathToWebAppSrcDir = [reldirP|./|]
+mkTmplDataForExtImport :: Maybe ExtImport -> Aeson.Value
+mkTmplDataForExtImport maybeExtImport = maybe notDefinedValue mkTmplData maybeExtImport
```

Comment:
This function encapsulates a common pattern.

If I understand correctly, we're using it for all ext imports. Do you think it makes sense to put it in a more public place and reuse?

Reply:
This can be extracted to a common function in `Wasp.Generator.JsImport` and then it can be specialized in `Wasp.Generator.WebAppGenerator.JsImport` and `Wasp.Generator.ServerGenerator.JsImport`.

In the web app templates, this pattern is actually only used in this file, but in the server templates it's used more.

I'll extract the pattern in this PR and apply it to `WebApp` only. I'll do the same for server templates in a separate PR.

Reply:
We now have:
- `Wasp.Generator.JsImport` -> `jsImportToTmplData` which generates approprisate `JSON` for usage in templates when we need to do something with import statements
- `Wasp.Generator.WebAppGenerator.JsImport` -> `extImportToJsImportTmplData` which uses the above method to convert an `ExtImport` to `JSON` template data

We should add in a separate PR:
- `Wasp.Generator.ServerGenerator.JsImport` -> `extImportToJsImportTmplData`
- use the new function to refactor the way use generate template data for JS import statements

Reply:
To keep in line with naming conventions for functions (i.e., they should be verbs) and what we already have (like this guy [here](https://github.com/wasp-lang/wasp/blob/868362d103469b9cfc6b1ed5b66c26eae7e33da5/waspc/src/Wasp/Generator/Common.hs#L58)*), I suggest renaming this to something like `makeImportJson`.

The "functions should be verbs" applies to other places as well.

\* Although that name could be improved too :)

Reply:
We have a number of these "tranform" functions in our codebase, and I kinda like this convetion of `<A>To<B>` naming.

Some examples I found in our codebase:
- `kebabToCamelCase`
- `sourceSpanToRegion`
- `lspPositionToOffset`
- `extImportToJsImport` (that's mine)

What about `extImportToImportJson`?

Reply:
That's true, we have several of those.

Anyway, choose whatever you like best.

Purely FYI, this is my order of preference:
1. `makeExtImportJson`
2. `extImportToImportJson`
3. `extImportToJsImport`